tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+        #if $overrepresented_sequence_analysis.overrepresentation_analysis:
+            $overrepresented_sequence_analysis.overrepresentation_analysis
+            #if str($overrepresented_sequence_analysis.overrepresentation_sampling):
+                -P $overrepresented_sequence_analysis.overrepresentation_sampling
+            #end if
+        #end if
+
+        ## thread options
+        --thread \${GALAXY_SLOTS:-1}
+        ]]>
+    </command>
+
+    <inputs>
+        <conditional name="single_paired">
+            <param name="single_paired_selector" type="select" label="Single-end or paired reads">
+                <option value="single" selected="true">Single-end</option>
+                <option value="paired">Paired</option>
+            </param>
+            <when value="single">
+                <param format="fastq" name="in1" type="data" label="Input 1" argument="-i" help="Input FastQ file #1."/>
+                <section name="adapter_trimming_options" title="Adapter trimming options" expanded="False">
+                    <param name="disable_adapter_trimming" type="boolean" truevalue="-A" falsevalue="" checked="false" label="Disable adapter trimming" argument="-A" help="Adapter trimming is enabled by default. If this option is specified, adapter trimming is disabled."/>
+                    <param name="adapter_sequence1" type="text" optional="true" label="Adapter sequence for input 1" argument="--adapter_sequence" help="The adapter for read1. For SE data, if not specified, the adapter will be auto-detected. For PE data, this is used if R1/R2 are found not overlapped.">
+                        <sanitizer>
+                            <valid>
+                                <add value="A"/>
+                                <add value="T"/>
+                                <add value="C"/>
+                                <add value="G"/>
+                            </valid>
+                         </sanitizer>
+                    </param>
+                </section>
+                <section name="global_trimming_options" title="Global trimming options" expanded="False">
+                    <param name="trim_front1" type="integer" optional="true" label="Trim front for input 1" argument="-f" help="Trimming how many bases in front for read1, default is 0."/>
+                    <param name="trim_tail1" type="integer" optional="true" label="Trim tail for input 1" argument="-t" help="Trimming how many bases in tail for read1, default is 0."/>
+                </section>
+            </when>
+            <when value="paired">
+                <param format="fastq" name="in1" type="data" label="Input 1" argument="-i" help="Input FastQ file #1."/>
+                <param format="fastq" name="in2" type="data" optional="true" label="Input 2" argument="-I" help="Input FastQ file #2."/>
+                <section name="adapter_trimming_options" title="Adapter trimming options" expanded="False">
```

Comment:
This section is a repeat of  lines 143-155, you can place this into a [macro](https://planemo.readthedocs.io/en/latest/writing_advanced.html#macros-reusable-elements) at the top and then expand the macro whenever you need to.

See an example [here](https://github.com/galaxyproject/tools-iuc/pull/1597/files#diff-2293f1039f66cfd118915415edaf6538R5), and the expansions [here](https://github.com/galaxyproject/tools-iuc/pull/1597/files#diff-2293f1039f66cfd118915415edaf6538R234) and [here](https://github.com/galaxyproject/tools-iuc/pull/1597/files#diff-2293f1039f66cfd118915415edaf6538R250)